From f26d40a3fedca74d4a7a34042b09d62ca8ed68e8 Mon Sep 17 00:00:00 2001
From: Edward Hervey <bilboed@bilboed.com>
Date: Sun, 5 Jan 2025 12:05:05 +0100
Subject: [PATCH] opcodes: Use MIN instead of CLAMP for known unsigned values

The same fix was already done for other times ages ago. These two were missing

Part-of: <https://gitlab.freedesktop.org/gstreamer/orc/-/merge_requests/212>
---
 orc/opcodes.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/orc/opcodes.h b/orc/opcodes.h
index b3c5a09..5b979a9 100644
--- a/orc/opcodes.h
+++ b/orc/opcodes.h
@@ -110,16 +110,16 @@ UNARY_LW(convlw, "%s")
 UNARY_WB(convhlw, "((orc_uint32)%s)>>16")
 UNARY_LW(convssslw, "ORC_CLAMP_SW(%s)")
 UNARY_LW(convsuslw, "ORC_CLAMP_UW(%s)")
-UNARY_LW(convusslw, "ORC_CLAMP((orc_uint32)%s,0,ORC_SW_MAX)")
-UNARY_LW(convuuslw, "ORC_CLAMP_UW((orc_uint32)%s)")
+UNARY_LW(convusslw, "ORC_MIN((orc_uint32)%s, ORC_SW_MAX)")
+UNARY_LW(convuuslw, "ORC_MIN((orc_uint32)%s, ORC_UW_MAX)")
 
 UNARY_LQ(convslq, "%s")
 UNARY_LQ(convulq, "(orc_uint32)%s")
 UNARY_LW(convql, "%s")
 UNARY_LW(convsssql, "ORC_CLAMP_SL(%s)")
 UNARY_LW(convsusql, "ORC_CLAMP_UL(%s)")
-UNARY_LW(convussql, "ORC_CLAMP_SL((orc_uint64)%s)")
-UNARY_LW(convuusql, "ORC_CLAMP_UL((orc_uint64)%s)")
+UNARY_LW(convussql, "ORC_MIN((orc_uint64)%s, ORC_SL_MAX)")
+UNARY_LW(convuusql, "ORC_MIN((orc_uint64)%s, ORC_UL_MAX)")
 
 BINARY_BW(mulsbw, "%s * %s")
 BINARY_BW(mulubw, "((orc_uint16)((orc_uint8)%s)) * ((orc_uint16)((orc_uint8)%s))")
-- 
2.45.2

